source/balances/contracts/BalanceChecker.sol

```diff
@@ -0,0 +1,218 @@
+/*
+    Utility contract for checking balances and allowances for ERC20 tokens
+    Some of these functions originated from (and then improved upon) DeltaBalances.github.io
+    Check balances for multiple ERC20 tokens for multiple users in 1 batched call
+    Contains an admin - owner to return any lost ether or tokens if necessary
+*/
+
+pragma solidity 0.5.12;
+
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "@airswap/tokens/contracts/interfaces/IWETH.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
+import "openzeppelin-solidity/contracts/utils/Address.sol";
+
+/**
+  * @title BalanceChecker: Contract for checking balances and allowances for ERC20 tokens
+  * @notice Supports checks for fungible tokens (ERC-20)
+  */
+contract BalanceChecker is Ownable {
+  using SafeERC20 for IERC20;
+  using Address for address;
+
+  /**
+    * @notice Check the token balances of a wallet for multiple tokens
+    * @dev return array and will fail if large token arrays are inputted
+    * @dev Returns array of token balances in base units
+    * @param userAddress address
+    * @param tokenAddresses address[]
+    * @return uint256[] token balance array if possible
+    */
+  function walletBalances(
+    address userAddress,
+    address[] calldata tokenAddresses
+  ) external view returns (uint256[] memory) {
+    require(tokenAddresses.length > 0);
+    uint256[] memory balances = new uint256[](tokenAddresses.length);
+
+    for(uint256 i = 0; i < tokenAddresses.length; i++){
+      if( tokenAddresses[i] != address(0x0) ) {
```

Comment:
weird spacing on the brackets. For `if` `while` and `for` we formally do:
`if (...) {`